bullet_train/app/helpers/account/dates_helper.rb

```diff
@@ -1,40 +1,24 @@
 module Account::DatesHelper
-  # e.g. October 11, 2018
-  def display_date(timestamp, custom_date_format = nil)
+  def display_date(timestamp, format: nil)
     return nil unless timestamp
-    if custom_date_format
-      local_time(timestamp).strftime(custom_date_format)
-    elsif local_time(timestamp).year == local_time(Time.now).year
-      local_time(timestamp).strftime("%B %-d")
-    else
-      local_time(timestamp).strftime("%B %-d, %Y")
-    end
+    timestamp = set_to_user_time_zone(timestamp)
+    formatted_timestamp(timestamp, format: format)
   end
+  alias_method :display_date_and_time, :display_date
 
-  # e.g. October 11, 2018 at 4:22 PM
-  # e.g. Yesterday at 2:12 PM
-  # e.g. April 24 at 7:39 AM
-  def display_date_and_time(timestamp, custom_date_format = nil, custom_time_format = nil)
-    return nil unless timestamp
-
-    # today?
-    if local_time(timestamp).to_date == local_time(Time.now).to_date
-      "Today at #{display_time(timestamp, custom_time_format)}"
-    # yesterday?
-    elsif (local_time(timestamp).to_date) == (local_time(Time.now).to_date - 1.day)
-      "Yesterday at #{display_time(timestamp, custom_time_format)}"
-    else
-      "#{display_date(timestamp, custom_date_format)} at #{display_time(timestamp, custom_time_format)}"
-    end
-  end
+  private
 
-  # e.g. 4:22 PM
-  def display_time(timestamp, custom_time_format = nil)
-    local_time(timestamp).strftime(custom_time_format || "%l:%M %p")
+  # If the format is a Symbol, we use the standard Rails formatting.
+  # https://guides.rubyonrails.org/i18n.html#adding-date-time-formats
+  def formatted_timestamp(timestamp, format: nil)
+    format ||= :default
+    format.is_a?(Symbol) ? l(timestamp, format: format) : timestamp.strftime(format)
   end
```

Comment:
It looks like `I18n.localize`, which Rails delegates to, already handles the Symbol lookup but falls back to a format string based to `strftime`. Have we tried writing this like this?

```suggestion
  def formatted_timestamp(timestamp, format: :default)
    localize(timestamp, format: format)
  end
```

However, if this works, at that point we could also inline the method.

Reply:
Thinking more about this, I wonder if we want these helpers to just be pretty close to this:

```ruby
module Account::DatesHelper 
  def display_date(timestamp, ...)
    localize(local_time(timestamp.to_date), ...) if timestamp
  end

  def display_time(timestamp, ...)
    localize(local_time(timestamp.to_time), ...) if timestamp
  end

  def display_datetime(timestamp, ...)
    localize(local_time(timestamp.to_datetime), ...) if timestamp
  end

  def local_time(timestamp)
    timestamp&.in_time_zone(current_user.time_zone || Time.zone)
  end
end
```

Reply:
Yes, this makes sense to me and is much cleaner to look at. I think I got bogged down in trying to use what was already there, but this is easier to read. I committed what you wrote, but I have an idea to take it a step further by putting all of the logic into one method which I'll mention in a new comment below.

Reply:
Oof, wasn't aware that it would fallback to `strtime`, just passing `format` will definitely work then. As far as the `format: :default` keyword argument, I originally tried writing it the way you mentioned here, but I eventually wrote `format ||= :default` here because `format` kept returning `nil` for me:

![image](https://user-images.githubusercontent.com/10546292/224952423-54a81490-5a54-4adc-876f-1f79f6549693.png)

I found out it was basically because if we explicitly pass `nil` to the method, the default keyword argument just gets overwritten with `nil`. This was happening because I was calling `display_date(timestamp, format: local_assigns[:format])` in the view which was passing `nil` if we don't give it a value when rendering the view.
```
irb(main):001:1* def foo(bar: true)
irb(main):002:1*   puts "Value: #{bar}"
irb(main):003:0> end
=> :foo
irb(main):004:0> foo
Value: true
=> nil                                             
irb(main):005:0> foo(bar: nil)
Value: 
=> nil
```

The keyword arguments should be working now!

Reply:
I think I spot a typo there @gazayas, shouldn't that be:
```
[1] pry(main)> def foo( bar = true)
[1] pry(main)*   puts "Value: #{bar}"
[1] pry(main)* end  
=> :foo
[2] pry(main)> foo
Value: true
=> nil
[3] pry(main)> 
```

Reply:
@kaspth Of course, the even shorter alias for localize is just 'l' (similar to translate and 't'), making this even neater still (though arguably harder to read?)

Reply:
I would suggest we avoid defaulting to current_user.time_zone in these.  If you look at the code for [current_locale_method](https://github.com/bullet-train-co/bullet_train-core/blob/7b6aa6e3dada6f7fb2fbb54e60cbb21772defe61/bullet_train/app/helpers/account/locale_helper.rb#L2), we check User > Team > default locale.  Unfortunately, that is hardcoded, which is shame (would recommend I18n.default_locale, as per Rails conventions).

However, the concept of the cascading priorities would seem to apply here as well, so perhaps:
```
  def current_time_zone
    current_user.time_zone || current_team.time_zone || Time.zone
  end
```

Would you agree?

Reply:
@gazayas yeah, that's an issue sometimes. I'd write them as `localize(local_time(timestamp.to_x), format: format || :default) if timestamp` then.

@kitebuggy I tend to prefer `l` for app code in ERB and then `localize` for lib code, but it's just personal preference really.

From what I can tell, `current_time_zone` sounds like it fits in this context and is interchangeable with our `current_user.time_zone || Time.zone` handling.